### ISSUE TITLE
Gmail Authorization Allowed to Expire - "Execution is running" -> "Problem stopping execution, The execution ID could not be found"

### ISSUE THREAD

**Issue by zikeji** [n8n-io/n8n#5878]
**Describe the bug**
We have a Gmail automation using polling. Yesterday it started failing but without actually "failing". Just spawning a bunch of "Running" executions every minute. We eventually narrowed the bug down to the Gmail authorization expiring (which shouldn't be possible as N8N should be refreshing the token - it was not invalidated from our end). Had to delete the executions from the database directly as clicking "Stop Execution" resulted in a not found error.

**To Reproduce**

1. Create a GMail Oauth2 API credential, set it up per documentation, authenticate an account
2. Create a new workflow, add a GMail trigger node with default settings
3. Send a test email to verify the execution went off properly (or don't, up to you)
4. Go to the Google Account's "Apps with access to your account" section and revoke authorization to your N8N credentials
5. Wait as you start seeing a stuck execution spawn every minute. Also try to delete them - no dice.

If you turn off the workflow and attempt to turn it back on you now get:
> There was a problem activating the workflow: "Cannot read properties of undefined (reading 'reduce')"

**Expected behavior**

The problem is twofold. For starts, we're not sure how - but the token appears to have been invalidated/expired and not refreshed properly. However, without a explanation that isn't the topic of this report and if I do come across reproduction steps I will create a new report.

If authorization fails for whatever reason, we would expect executions to not get stuck in the running state, impossible to remedy, and instead expect them to get marked as failed with an error reflecting why. In my test environment, my error workflow still triggered but with a blank error and no flavor text, despite the execution stuck in the running state. The stuck executions also shouldn't be inappropriately reporting they don't exist when trying to delete, and instead allow themselves to be deleted. It shouldn't require a `DELETE FROM` SQL statement to rectify.

**Environment (please complete the following information):**

- **OS:** Alpine Linux v3.17 (Docker image)
- **n8n Version:** 0.222.0
- **Node.js Version:** v16.19.1
- **Database system:** 10.9.4-MariaDB-1:10.9.4+maria~ubu2204
- **Operation mode:** own?

**Additional context**

Interesting to note, when I was originally trying to reproduce I didn't realize my test instance was `0.214.0` - on 0.214.0 the executions did not get stuck in the running state, but the error text was still empty/nil.

**Comment by Joffcom** [comment]
Hey @zikeji 

We are aware of an issue with the google oauth tokens expiring and not refreshing.

At the moment the best solution is to downgrade to 221.2 or 220.1, we pulled the 222.0 release yesterday afternoon to minimise the impact.

**Comment by zikeji** [comment]
@Joffcom Thanks for the response!

We originally experienced this issue on 221.2 (the deauth and the executions not transitioning to error state), I only tested / created reproduction steps for 222.0 as it was the latest.

Can I assume this also means you're aware of the executions getting stuck in the running state and being unable to delete them?

**Comment by Joffcom** [comment]
Hey @zikeji 

I have seen a couple of reports of executions getting stuck but have not dug into that yet.

**Comment by ldaume** [comment]
I am having the same problem with Gmail. The latest working version is 222.1, since then I get an Authorization failed error.

ERROR: Authorization failed - please check your credentials (item 0)
Request failed with status code 401

```
NodeOperationError: Authorization failed - please check your credentials (item 0)
    at Object.execute (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/nodes/Google/Gmail/v2/GmailV2.node.ts:800:11)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at Workflow.runNode (/usr/local/lib/node_modules/n8n/node_modules/n8n-workflow/src/Workflow.ts:1261:19)
    at /usr/local/lib/node_modules/n8n/node_modules/n8n-core/src/WorkflowExecute.ts:949:29
```

I hope to find time to do some bugfixing in the next few days 😉 .

**Comment by Joffcom** [comment]
Hey @ldaume,

`0.222.1` should contain the fix but can you try `223` and see if that works for you, As a test if you downgrade to 220.1 does it start working without needing to be connected again?

**Comment by ldaume** [comment]
This is confusing @Joffcom. I had updated to latest and the problem was still there, and now I explicitly updated to version 0.223.0, and it seems to work again. 🎉

**Comment by Joffcom** [comment]
Hey @ldaume,

That is odd but I am glad it is working. I think all of the issues reported in this thread should be resolved in 223

**Comment by Joffcom** [comment]
Looks like I forgot to mark this as closed, The main issues were resolved in 0.223.0.
